src/components/Table/nestedCards.tsx

```diff
@@ -0,0 +1,177 @@
+    div = (
+      <div
+        css={{
+          ...Css.bgColor(card.bgColor).pxPx(card.pxPx).$,
+          // Only the 1st div needs border radius.
+          ...(!div &&
+            Css.add({
+              [`border${place}RightRadius`]: `${card.brPx}px`,
+              [`border${place}LeftRadius`]: `${card.brPx}px`,
+            }).hPx(card.brPx).$),
+          ...(card.bColor && Css.bc(card.bColor).bl.br.if(div)[btOrBb].$),
+        }}
+      >
+        {div}
+      </div>
+    );
+  });
+  return div;
+}
+
+/**
+ * For the first or last cell of actual content, wrap them in divs that re-create the
+ * outer cards' padding + background.
+ */
+export function maybeAddCardPadding(
+  columns: GridColumn<any>[],
+  openCards: NestedCardStyle[],
+  idx: number,
+  div: any,
+): any {
+  const isFirst = idx === 0;
+  const isFinal = idx === columns.length - 1;
+  if (!isFirst && !isFinal) {
+    return div;
+  }
+  [...openCards].reverse().forEach((card) => {
+    div = (
+      <div
+        css={{
+          ...Css.bgColor(card.bgColor).if(!!card.bColor).bc(card.bColor).$,
+          ...(isFirst && Css.plPx(card.pxPx).if(!!card.bColor).bl.$),
+          ...(isFinal && Css.prPx(card.pxPx).if(!!card.bColor).br.$),
+        }}
+      >
+        {div}
+      </div>
+    );
+  });
+  return div;
+}
+
+/** Create a spacer between rows of children. */
+export function makeSpacer(current: NestedCardStyle, openCards: NestedCardStyle[]) {
+  let div = <div css={Css.hPx(current.spacerPx).$} />;
+  // Start at the current/inside card, and wrap outward padding + borders.
+  // | card1 | card2 | ... card3 ... | card2 | card1 |
+  [...openCards].reverse().forEach((card) => {
+    div = <div css={Css.bgColor(card.bgColor).pxPx(card.pxPx).if(!!card.bColor).bc(card.bColor).bl.br.$}>{div}</div>;
+  });
+  return div;
+}
+
+/**
+ * Takes the current buffer of close row(s), spacers, and open row, and creates a single chrome DOM row.
+ *
+ * This allows a minimal amount of DOM overhead, insofar as to the css-grid or react-virtuoso we only
+ * 1 extra DOM node between each row of content to achieve our nested card look & feel, i.e.:
+ *
+ * - chrome row (open)
+ * - card1 content row
+ * - chrome row (card2 open)
+ * - nested card2 content row
+ * - chrome row (card2 close, card1 cloard)
+ */
+export function maybeCreateChromeRow(
+  columns: GridColumn<any>[],
+  filteredRows: RowTuple<any>[],
+  chromeBuffer: JSX.Element[],
+): void {
+  if (chromeBuffer.length > 0) {
+    filteredRows.push([
+      undefined,
+      <div css={Css.gc(`span ${columns.length}`).$}>
+        {chromeBuffer.map((c, i) => (
+          <Fragment key={i}>{c}</Fragment>
+        ))}
+      </div>,
+    ]);
+    // clear the buffer
+    chromeBuffer.splice(0, chromeBuffer.length);
+  }
+}
+
+export function dropChromeRows<R extends Kinded>(filteredRows: RowTuple<R>[]): [GridDataRow<R>, ReactElement][] {
+  return filteredRows.filter(([r]) => !!r) as [GridDataRow<R>, ReactElement][];
```

Comment:
**question:** Is this due to all chrome rows being of `kind` `undefined`?

Reply:
Basically yes; technically there is not _kind_ `undefined` per se, but yes they don't have a `GridDataRow` so the tuples of `filteredRows` used to be:

* `[row1, <RowComponent />`
* `[row2, <RowComponent />`
* `[row3, <RowComponent />`
* `[row4, <RowComponent />`

Are now:

* `[undefined, <chrome row/>`
* `[row1, <RowComponent />`
* `[undefined, <chrome row />`
* `[row2, <RowComponent />`
* ...

So yeah, this strips out the extra rows, for places that want to still see the real/original rows.